task/activity.go

```diff
@@ -52,6 +54,10 @@ type activityContext struct {
 type Activity func(ctx ActivityContext) (any, error)
 
 func newTaskActivityContext(ctx context.Context, taskID int32, ts *protos.TaskScheduledEvent) *activityContext {
+	ctx, err := helpers.ContextFromTraceContext(ctx, ts.ParentTraceContext)
```

Comment:
I would expect that the `ctx` argument already has the correct trace context, being set much earlier than this.

Reply:
Yes, for examples running not through the grpc we did have the trace context, but if we run it through grpc we are missing it.

Reply:
We can actually set the context in `processWorkItem` itself: https://github.com/microsoft/durabletask-go/blob/fc5756737d3c85ac4903adf35b58fce25f36641b/client/worker_grpc.go#L92